src/stackAnalysis.ts

```diff
@@ -44,25 +45,63 @@ export class Stack {
     }
 
     public dump(): string {
-        return this._stack.map(se => se.name).join(', ');
+        return this._stack
+            .map(item => 
+                item.type === "guard" 
+                    ? `GUARD(${item.name})` // Customize guard display
+                    : item.name // Display regular variable names
+            )
+            .join(", ");
     }
-
+    
     public pop(): StackVariable {
         let result = this._stack.pop();
         if (result == undefined) {
             throw new StackUnderflowError(1);
+        } else if (result.type === "guard") {
+            throw new Error("Stack pop attempted on a guard. Use pop_guard instead.");
```

Comment:
тут чатгпт зря подсказку сделал, если мы пытаемся попнуть гард, то это скорее стрёмный контракт, чем недоработка декомпилятора. Ещё хорошо бы сюда добавить информацию о том, какой именно гард мы пытаемся попнуть (его метадату сдампить)

```suggestion
            throw new Error("Stack pop attempted on a guard");
```